pkg/forwarder/transaction.go

```diff
@@ -212,18 +212,25 @@ type Transaction interface {
 	GetPriority() TransactionPriority
 	GetEndpointName() string
 	GetPayloadSize() int
+	SerializeTo(*TransactionsSerializer) error
```

Comment:
I forgot to add a comment explaining this method. 
Note: I do not think `Profobuf` should be in the name of the method as it is an implementation details.

Reply:
Comment added: 
https://github.com/DataDog/datadog-agent/blob/74f2993307b94450b827271cb57ebbb75aa4237c/pkg/forwarder/transaction.go#L216-L218.
It is a variation of the [Visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern).

Reply:
Yes but `Transaction` doesn't actually now how to serialize itself since it call `Add` on `TransactionsSerializer`. `TransactionsSerializer` hold the knowledge.

Reply:
Comment updated.

Reply:
I'm still not convince it's useful `Transaction` being in the public API we should avoid requiring method if we don't have to.

Reply:
The alternative solution I have in mind is casting the collection of `Transaction` to a collection of `HTTPTransaction` before the serialization. If the type is not `HTTPTransaction`, an error is logged. The drawback of this solution is that 
a new implementation of `Transaction` is likely to not update this code as, in a normal usage, the retry queue is not written on disk and so this error is not logged. In my experience this kind of code worths it over the time.

Maybe you have a better alternative solution in mind?
Anyway, if you feel it is useless, I can remove this code.